src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -258,6 +260,16 @@ export class TerminalService implements ITerminalService {
 				? this._localTerminalsInitPromise = this._reconnectToLocalTerminals()
 				: Promise.resolve();
 		this._primaryOffProcessTerminalService = !!this._environmentService.remoteAuthority ? this._remoteTerminalService : this._localTerminalService;
+		this._primaryOffProcessTerminalService.onDidRequestDetach(async (e) => {
+			if (e.workspaceId && this._workspaceContextService.getWorkspace().id === e.workspaceId) {
+				const instanceToDetach = this.getInstanceFromId(e.instanceId);
+				const persistentProcessId = instanceToDetach?.persistentProcessId;
+				if (persistentProcessId && !instanceToDetach.shellLaunchConfig.isFeatureTerminal && !instanceToDetach.shellLaunchConfig.customPtyImplementation) {
+					await instanceToDetach.detachFromProcess();
+					await this._primaryOffProcessTerminalService?.acceptDetachedInstance(e.requestId, persistentProcessId);
+				}
```

Comment:
Need an else case:

```ts
} else {
  this._primaryOffProcessTerminalService?.acceptDetachedInstance(e.requestId, undefined);
}
```

A comment explaining what undefined means here would be good too